### ISSUE TITLE
Support TypeScript 2.x.

### ISSUE THREAD

**Issue by asial-matagawa** [monaca/monaca-lib#90]
@andipavllo @frankdiox 

Reported by @anatoo.
Since 1.0.0-rc2, `angular2-onsenui` requires TypeScript 2.x but current monaca-lib is using TypeScript 1.8.10 as a build dependency.
So `monaca-lib` needs to update `typescript` in `package.json` from `1.8.10` to `2.0.10` (latest).

But I am not sure that it is enough, so could you update `typescript` in right way?

Note:
- Current Angular 2 monaca-templates are also incompatible with TypeScript 2.x.
  We have to fix Angular 2 monaca-templates **before updating monaca-lib**.

**Comment by frandiox** [comment]
@asial-matagawa I can update monaca-lib but I don't know too much about TypeScript templates. Can you please update those? Or maybe they are already updated?

**Comment by asial-matagawa** [comment]
@frankdiox 
>Can you please update those?

Sure, I will update those.

>Or maybe they are already updated?

No, the current version (`2.2.2+20161205.1`) is incompatible with TypeScript 2.x.

----
`Current Angular 2 monaca-templates are also incompatible with TypeScript 2.x.` actually means that simply a typings d.ts file in the templates  ( https://github.com/monaca-templates/onsenui-v2-angular2-minimum/blob/master/typings/globals/node/index.d.ts ) is outdated.
It causes an error like:
```
ERROR in /Users/asial/temp/asdfasdf/typings/globals/node/index.d.ts
(384,11): error TS2430: Interface 'NodeBuffer' incorrectly extends interface 'Uint8Array'.
  Types of property 'fill' are incompatible.
    Type '(value: any, offset?: number, end?: number) => Buffer' is not assignable to type '(value: umber, start?: number, end?: number) => this'.
      Type 'Buffer' is not assignable to type 'this'.
Project has failed to transpile. Error has occured while transpiling /Users/asial/temp/asdfasdf with webpack. Please check the logs.
```

Maybe what we should do about Angular 2 monaca-templates is only to update the typings d.ts files.

**Comment by asial-matagawa** [comment]
@frankdiox 
I have updated the four Angular 2 templates:
https://github.com/monaca-templates/onsenui-v2-angular2-minimum
https://github.com/monaca-templates/onsenui-v2-angular2-navigation
https://github.com/monaca-templates/onsenui-v2-angular2-splitter
https://github.com/monaca-templates/onsenui-v2-angular2-tabbar
The new versions of the templates are incompatible with monaca-lib 2.2.2 and 2.2.3, so I created a new tag `2.2.4`.

So, what I would like ask you to do is the following changes:
- Update `typescript` from `1.8.10` to `2.0.10`.
    - [Caution] If you set something like `^2.0.0`, `monaca reconfigure` will be broken.
- Update `ts-loader` from `0.8.2` to `1.3.0`.
    - [Caution] ts-loader 0.8.2 is incompatible with TypeScript 2.x.
- Update CHANGELOG.

Thank you.

**Comment by asial-matagawa** [comment]
@frankdiox @andipavllo 
I have set up CircleCI settings of the templates so that you can test them with HEAD branch of CLI and lib:
https://circleci.com/gh/monaca-templates/onsenui-v2-angular2-minimum/edit#setup
https://circleci.com/gh/monaca-templates/onsenui-v2-angular2-navigation/edit#setup
https://circleci.com/gh/monaca-templates/onsenui-v2-angular2-splitter/edit#setup
https://circleci.com/gh/monaca-templates/onsenui-v2-angular2-tabbar/edit#setup

In other words, you can make sure whether or not new CLI and lib will work on the templates before releasing CLI and lib.

Could you release CLI 2.1.7 and lib 2.2.4 according to https://github.com/monaca/monaca-lib/issues/90#issuecomment-265444579 if you have time?
New CLI and lib are required by:
- angular2-onsenui 1.0.0-rc.2 and later
- https://github.com/OnsenUI/OnsenUI/issues/1728
- https://github.com/OnsenUI/OnsenUI/issues/1737

**Comment by asial-matagawa** [comment]
@frankdiox 
Thank you for releasing CLI and lib!
Closing This.
